compiler/rustc_const_eval/src/interpret/step.rs

```diff
@@ -250,8 +254,13 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 let src = self.eval_place(place)?;
                 let place = self.force_allocation(&src)?;
                 let mut val = ImmTy::from_immediate(place.to_ref(self), dest.layout);
-                if !place_base_raw {
+                if !place_base_raw
+                    && span.desugaring_kind() != Some(DesugaringKind::IndexBoundsCheckReborrow)
```

Comment:
Wasn't the point to get rid of this?

Reply:
You are reviewing the wrong code

Reply:
That is to say, you're reviewing the *un-revert commit*, not the functional changes on top.

Reply:
Oh, maybe add a warning that commit-by-commit review doesn't work here... this would have been easier to review if there was one commit just adding `RawRefKind` without any functional changes. It seems instead the functional changes are burried in a >1k line patch?

Reply:
I also don't understand the point of having the un-revert commit, if it is being re-reverted in the next commit.

Reply:
It's not being reverted. The commit I stacked on top changes a small portion of the code dealing with the raw ref op that we emit as part of the lowering.

Reply:
Specifically, I stacked the code this way so it's very clear what code was already committed versus what I wrote.